engine/src/main/java/org/terasology/rendering/nui/editor/utils/NUIEditorMenuTreeBuilder.java

```diff
@@ -250,7 +252,7 @@ private void populateContextMenu(JsonTree node, MenuTree addTree, boolean isSkin
                 // If the node is part of a layout, add an option to add the layoutInfo node.
                 if (nodeInfo.getLayoutClass() != null) {
                     String layoutInfo = "layoutInfo";
-                    if (!node.hasChildWithKey(layoutInfo)) {
+                    if (!Objects.equals(node.getValue().getKey(), layoutInfo) && !node.hasChildWithKey(layoutInfo)) {
```

Comment:
Couldn't manage to retrieve `layoutHintType` - for some odd JVM-y reason it resolves to null every time. All the other criteria should now be met, though.